README.md

```diff
@@ -8,6 +8,77 @@ https://github.com/ansible/awx/tree/devel/awx/main/dispatch
 As a part of doing the split, we also want to resolve a number of
 long-standing design and sustainability issues, thus, asyncio.
 
+### Usage
+
+You have a postgres server configured and a python project.
+You will use dispatcher to trigger a background task over pg_notify.
+
+Both your *background dispatcher service* and your *task publisher* process must have
+python configured so that your task is importable.
+
+#### Library
+
+The dispatcher `@task()` decorator is used to register tasks.
+
+See the `tools/test_methods.py` module.
+This defines a dispatcher task and the pg_notify channel it will be sent over.
+
+```python
+from dispatcher.publish import task
+
+@task(queue='test_channel')
+def print_hello():
+    print('hello world!!')
+```
+
+#### Dispatcher service
+
+The dispatcher service needs to be running before you submit tasks.
+This does not make any attempts at message durability or confirmation.
+
+There are 2 ways to run the dispatcher:
+ - Importing and running
```

Comment:
Lets add a snippet as example to run the process, WDYT?
```suggestion
 - Importing and running
  ```python
  from dispatcher.main import DispatcherMain
  import asyncio
  
  pgnotify_dsn = "dbname=postgres user=postgres"
  config = {
      "producers": {
          "brokers": {
              "pg_notify": {"conninfo": pgnotify_dsn},
              "channels": [
                  "test_channel",
              ],
          },
      },
      "pool": {"max_workers": 4},
  }
  loop = asyncio.get_event_loop()
  dispatcher = DispatcherMain(config)
  
  
  try:
      loop.run_until_complete(dispatcher.main())
  finally:
      loop.close()
  ```
```

Reply:
I was looking over this README and noticed the same thing. It gives code for the definition and publisher, but not the service. So agree.